File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Ordenamientos/Count_Sort.java
import java.util.Arrays;
import java.util.Iterator;

//COUNT_SORT
public class Ordenamientos2_4 {

	static void countSort(char arr[]) {
		int n = arr.length;
		char output [] = new char [n];
		int count [] = new int [256];
		//tamaño de 256 caracteres que hay en ASCII extendido
		Arrays.fill(count, 0);
		//Llena el vector de ceros
		for (int i = 0; i < n; i++) {
			count[arr[i]]++;
		}
		for (int i = 1; i <= 255; i++) {
			count[i] += count[i-1];
		}
		for (int i = n-1; i >= 0; i--) {
			output[count[arr[i]]-1] = arr[i];
			count[arr[i]]--;
		}
		for (int i = 0; i < n; i++) {
			arr[i] = output[i];
		}
	}
	
	
	
	
	public static void main(String[] args) {
		
		char arr[] = {'g', 'e', 'e', 'k', 'l', 'i', 'B', 'g', 'a', 'x', 
				'f', 'A', '@', '*'};
		countSort(arr);
		for (int i = 0; i < arr.length; i++) {
			System.out.print(arr[i] + " ");
		}
	}

}

File: Codigos estudiantes por lenguaje/C++/Daniel Cangrejo/Ordenamientos/Count_Sort.cpp
#include<bits/stdc++.h>
#include<cstdlib>
using namespace std;
void sortArr(char arr[],int n){
    char output[n];
    int Count[256];
    memset(Count,0,sizeof Count);
    for(int i=0;i<n;i++){
        Count[arr[i]]++;
    }
    for(int i=1;i<255;i++){
        Count[i]+=Count[i-1];
    }
    for(int i=n-1;i>=0;i--){
        output[Count[arr[i]]-1]=arr[i];
        --Count[arr[i]];
    }
    for(int i=0;i<n;i++){
        arr[i]=output[i];
    }
}
void printArr(char arr[],int n){
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}
int main() {
    char arr[]={'g','e','e','k','s','f','o','r','g','e','e','k','s'};
    int len=sizeof(arr)/sizeof(arr[0]);
    sortArr(arr,len);
    cout<<"caracteres ordenados"<<endl;
    printArr(arr,len);
}
